### PR TITLE
Update TigerRag Langchain Folder to Make it Match Newest Movie_Recs Logic #13

### PR DESCRIPTION
**Pull request by gjyotin305**
@tigerlab-ai 

All previously used helper functions 
- `generate_answer_with_rag_gpt3`
- `generate_answer_with_gar_gpt3`

They have been replaced with the modularized functions

### CODE REVIEW

**Comment by gjyotin305** [comment]
@wendyran

File: TigerRag/demos/langchain/demo.py [hunk 1]
```diff
@@ -2,17 +2,28 @@
...
 # Please set env var OPENAI_API_KEY for GAR and RAG.
 
 # Sample usage:
 #    python demo.py
 #    python demo.py -number_of_run 4
 
+# Initialize BERT tokenizer and model
+trag_bert_model = TigerRAGEmbeddingModel(EmbeddingModel.BERT)
+
+embedding_func = SentenceTransformerEmbeddings(model_name="bert-base-uncased")
```
Comment (tigerlab-ai) [inline TigerRag/demos/langchain/demo.py]: It seems this variable is not used anywhere.

File: TigerRag/demos/langchain/demo.py [hunk 2]
```diff
@@ -98,20 +117,40 @@ def main():
...
 
         # Example usage of GAR
         print("The following is GAR output for question: "+question)
-        print(generate_answer_with_gar_gpt3(
-            question, retrieved_context, 'text-davinci-003', vectorstore))
+        # print(generate_answer_with_gar_gpt3(
+        #     question, retrieved_context, 'text-davinci-003', vectorstore))
+        prompt_gar = prompt = f"Expand on the query: {question}"
+        openai_generative_query_augmenter = OpenAIQueryAugmenter(
+            "text-davinci-003")
```
Comment (tigerlab-ai) [inline TigerRag/demos/langchain/demo.py:85]: This seems to be duplicate of the code you updated in generate_answer_with_gar_gpt3()

File: TigerRag/demos/langchain/demo.py [hunk 3]
```diff
@@ -98,20 +117,40 @@ def main():
...
         # Example usage of RAG
         print("The following is RAG output for question: "+question)
-        print(generate_answer_with_rag_gpt3(
-            question, retrieved_context, 'text-davinci-003'))
+        # print(generate_answer_with_rag_gpt3(
+        #     question, retrieved_context, 'text-davinci-003'))
+
+        # Retrivel Augmented Generation
+        prompt_rag = f"Context: {retrieved_context} Question: {question}. Provide a summary or answer:"
+
```
Comment (tigerlab-ai) [inline TigerRag/demos/langchain/demo.py]: This seems to be duplicate with the code you updated in generate_answer_with_rag_gpt3(). Maybe remove generate_answer_with_rag_gpt3()?

File: TigerRag/demos/langchain/demo.py [hunk 4]
```diff
@@ -98,20 +117,40 @@ def main():
     for index in range(min(num_of_run, len(data["queries"]))):
         question = data["queries"][index]
         # Example usage of EBR
+        loader = WebBaseLoader(documents)
+        text_splitter = RecursiveCharacterTextSplitter(
+            chunk_size=500, chunk_overlap=0)
+        splits = text_splitter.split_documents(loader.load())
```
Comment (tigerlab-ai) [inline TigerRag/demos/langchain/demo.py:59]: This variable doesn't seem to be used/referenced anywhere.

**Review by tigerlab-ai — CHANGES_REQUESTED** [review]
Left a few comments. Overall looks good. Thanks!

**Comment by tigerlab-ai** [comment]
It seems your second commit address all the comments. Let me merge it. Thanks @gjyotin305

**Pull request merged**